sdk/communication/AzureCommunication/Tests/ObjCCommunicationTokenCredentialAsyncTests.m

```diff
@@ -106,10 +115,18 @@ - (void)xtest_ObjCRefreshTokenProactivelyTokenExpiringInNineMin {
         XCTAssertEqual(accessToken.token, weakSelf.sampleToken);
         XCTAssertEqual(weakSelf.fetchTokenCallCount, 1);
         
-        [expectation fulfill];
+        isComplete = YES;
     }];
 
-    [self waitForExpectations:@[expectation] timeout:2.0];
+    
+    NSDate *loopUntil = [NSDate dateWithTimeIntervalSinceNow: self.timeout];
+    while (isComplete == NO && [loopUntil timeIntervalSinceNow] > 0) {
```

Comment:
does this mean we will keep looping for 18 sec? if the flag isComplete is set the Yes, the test passed?

Reply:
Yes.

Reply:
inside the while loop, does any of the function is a blocking function? mean within this 18 sec will the CPU goes up to 100% inside this while loop?

Reply:
When `isComplete = YES` it will break out of the loop. Or if the `loopUntil` has exceeded the timeout